### DISCUSSION TITLE
feat: Add download as CSV to Data grid

### DISCUSSION THREAD

**Discussion by Ogglas** [microsoft/fluentui-blazor#441] [Ideas]
# 🙋 Feature Request

It would be great with a CSV export feature for data grid.  See MUI as an example: https://mui.com/x/react-data-grid/export/

## 🤔 Expected Behavior

Button named export that exports headers and values into a CSV file.

**Comment by vnbaaij** [comment]
I don't think this is something we would make available out of the box. But I'll turn this into a discussion on the form so we can gather more feedback

**Comment by mattgoedde** [comment]
In my opinion, I would just use the Toolbar component over the top of a DataGrid with all the actions I'd want.

**Comment by Swordancer** [comment]
I also think this is a good idea. Most of the available components offer grid export in various formats. If the grid can be sorted, filtered, manipulated (including hiding columns), it would become a very handy tool for quick report generation that can be exported to Excel or PDF. I understand this will probably require a lot of work, but to be honest, I prefer these components over the current commercial solutions.
